src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/jira/Jira.kt

```diff
@@ -16,9 +18,15 @@ class Jira(
     val address: URI,
     val jmxClients: List<JmxClient> = emptyList()
 ) : MeasurementSource {
+    private val logger: Logger = LogManager.getLogger(this::class.java)
+
     override fun gatherResults() {
+        if (nodes.isEmpty()) {
+            logger.warn("No Jira nodes known to JPT, not downloading node results")
+            return
+        }
         val executor = Executors.newFixedThreadPool(
-            Math.min(nodes.size, 4),
+            nodes.size.coerceAtMost(4),
```

Comment:
Ah yeah, our ancient non-idiomatic premature microoptimization :)
Alternatively, we could simplify away the thread pool size control (and the `if`) via `ExecutorService.newCachedThreadPool`